<launch>
<arg name="verbose" default="false"/>
<arg name="generate_points" default="true"/>
<arg name="cameras" default="[0,]" />
<arg name="sensor_model" default="helios"/>
<arg name="voxel_leaf" value="0.01" />
<arg name="filename" default=""/>
<arg name="rgb" default="aligned/rgb_to_depth/image_raw"/>
<arg name="depth" default="helios2/depth/image_raw"/>
<arg name="info" default="helios2/camera_info"/>
<arg name="rviz_name" default="unet.rviz"/>

<arg name="weight_file" value="$(find ros_unet)/obb_dataset_train/iternet_0.pth"/>
<!--arg name="weight_file" value="$(find ros_unet)/iternet_0_backup.pth"/-->

<node name="unet_server" pkg="ros_unet" type="ros_unet_server.py" output="screen" required="true">
    <rosparam subst_value="True">
      input_ch: 6
      weight_file: "$(arg weight_file)"
    </rosparam>
</node>

<node name="pipeline" pkg="ros_unet" type="obb_server" output="screen" required="true">
    <rosparam subst_value="True">
      verbose: $(arg verbose)
      min_points_of_cluster: 5
      sensor_model: $(arg sensor_model)
      voxel_leaf: $(arg voxel_leaf)
      generate_points: $(arg generate_points)
    </rosparam>
</node>

<node name="floor_detector" pkg="ros_unet" type="floor_detector" output="screen" required="true"/>
<node name="rosbag" pkg="rosbag" type="play" args="-r 0.5 --loop $(arg filename)" required="true" unless="$(eval filename=='')"/>

<!--node name="evaluator"  pkg="ros_unet" type="ros_eval.py" output="screen" required="true">
    <remap from="~PredictEdge" to="/unet_server/PredictEdge" />
    <remap from="~SetCamera"   to="/pipeline/SetCamera" />
    <remap from="~ComputeObb"  to="/pipeline/ComputeObb" />
</node-->

<node name="ros_client"  pkg="ros_unet" type="ros_client.py" output="screen" required="true">
    <remap from="~PredictEdge" to="/unet_server/PredictEdge" />
    <remap from="~SetCamera"   to="/pipeline/SetCamera" />
    <remap from="~ComputeObb"  to="/pipeline/ComputeObb" />
    <remap from="~FloorDetector/SetCamera"  to="/floor_detector/SetCamera" />
    <remap from="~FloorDetector/ComputeFloor"  to="/floor_detector/ComputeFloor" />
    <remap from="~cam0/rgb"    to="/cam0/$(arg rgb)" />
    <remap from="~cam0/depth"  to="/cam0/$(arg depth)" />
    <remap from="~cam0/info"   to="/cam0/$(arg info)" />
</node>


<node name="rviz" pkg="rviz" type="rviz" args="-d $(find ros_unet)/rviz/$(arg rviz_name)" required="true"/>
<arg name="load_config" default="true"/>
<arg name="side" default="down"/>
<group if="$(arg load_config)">
  <node pkg="rostopic" type="rostopic" name="tilt"
  args="pub /unloader/cam_tilt unloader_msgs/MultiarmPose 
  '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: ''},
    cam0_tilt: true,
    cam1_tilt: true}' -r 10" required="true" if="$(eval side=='up')"/>
  <node pkg="rostopic" type="rostopic" name="tilt"
  args="pub /unloader/cam_tilt unloader_msgs/MultiarmPose 
  '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: ''},
    cam0_tilt: false,
    cam1_tilt: false}' -r 10" required="true" if="$(eval side=='down')"/>
  <node pkg="tf2_ros" type="static_transform_publisher" name="link0"
  args="0 0 0 0 0 0 1 cam0_rgb_link cam0_rgb_camera_link"/>
  <node pkg="tf2_ros" type="static_transform_publisher" name="link1"
  args="0 0 0 0 0 0 1 cam1_rgb_link cam1_rgb_camera_link"/>
</group>
<include file="$(find unloader_calib)/launch/cam_tf_publisher.launch" if="$(arg load_config)"/>
<rosparam command="load" file="$(find ros_unet)/config/calib_info_for_tf.yaml" if="$(arg load_config)"/>


</launch>
